proxy/proxy.go

```diff
@@ -1151,6 +1145,7 @@ func (p *Proxy) do(ctx *context) (err error) {
 				rsp, perr2 = p.makeBackendRequest(ctx, backendContext)
 				if perr2 != nil {
 					p.log.Errorf("Failed to retry backend request: %v", perr2)
+					p.makeErrorResponse(ctx, perr2)
```

Comment:
Lets move this close to the return perr2

Reply:
done